### PR TITLE
Update MAPseq and add seed option

### PR DESCRIPTION
**Pull request by RZ9082**
FOR CONTRIBUTOR:
* [x] I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] License permits unrestricted use (educational + commercial)
* [ ] This PR adds a new tool or tool collection
* [x] This PR updates an existing tool or tool collection
* [ ] This PR does something else (explain below)

MAPseq 2.1.1b includes a `-seed` flag to ensure reproducibility.

### CODE REVIEW

File: tools/mapseq/mapseq.xml [hunk 1]
```diff
@@ -38,9 +38,11 @@
             ln -s '${ref_db.mscluster}' db.fasta.mscluster &&
             ln -s '${ref_db.mapseq2biom.otu_table}' db.otu &&
         #end if
-
         mapseq
         -nthreads \${GALAXY_SLOTS:-8}
+        #if $seed
```
Comment (bernt-matthias) [inline tools/mapseq/mapseq.xml:43]: With this you can not set `0` as a seed (which might/might not be fine).

**Review by bernt-matthias — COMMENTED** [review]
Should we use the parameter in out tests?

File: tools/mapseq/mapseq.xml (hunk 1 above)
Comment (RZ9082) [reply tools/mapseq/mapseq.xml:43]: Oh, sure. thanks :)

**Review by bernt-matthias — APPROVED** [review]

**Review by bernt-matthias — COMMENTED** [review]
Just to be sure, could you add a command line assertion to a test with (and without seed) checking that the `--seed` parameter is set (not set)?

File: tools/mapseq/mapseq.xml [hunk 2]
```diff
@@ -38,9 +38,11 @@
             ln -s '${ref_db.mscluster}' db.fasta.mscluster &&
             ln -s '${ref_db.mapseq2biom.otu_table}' db.otu &&
         #end if
-
         mapseq
         -nthreads \${GALAXY_SLOTS:-8}
+        #if $seed != ""
```
Comment (bernt-matthias) [inline tools/mapseq/mapseq.xml:43]:
```suggestion
        #if str($seed) != ""
```

File: tools/mapseq/mapseq.xml [hunk 3]
```diff
@@ -154,6 +158,9 @@
             <param name="mscluster" value="mapseq_db/LSU_trimmed.fasta.mscluster"/>
             <param name="mapseq2biom" value="no"/>
             <output name="classifications" file="sequences.mapseq" sort="true"/>
+            <assert_command>
+                <has_text text="-seed '12'" n="0" />
```
Comment (bernt-matthias) [inline tools/mapseq/mapseq.xml:162]: Just checking that the seed parameter is not set at all might be a bit better. 

```suggestion
                <has_text text="-seed" negate="true" />
```

File: tools/mapseq/mapseq.xml [hunk 4]
```diff
@@ -162,6 +169,9 @@
             <param name="taxonomy" value="mapseq_db/slv_lsu_filtered2_trimmed.txt"/>
             <param name="mapseq2biom" value="no"/>
             <output name="classifications" file="sequences.mapseq" sort="true"/>
+            <assert_command>
+                <has_text text="-seed '12'" n="0" />
```
Comment (bernt-matthias) [inline tools/mapseq/mapseq.xml:173]:
```suggestion
                <has_text text="-seed" negate="true" />
```

File: tools/mapseq/mapseq.xml [hunk 5]
```diff
@@ -189,13 +203,19 @@
             <output name="otu_tsv" file="mapseq2biom/tab-output_including_taxIDs.tabular" />
             <output name="otu_tsv_notaxid" file="mapseq2biom/tab-output_without_taxIDs.tabular" />
             <output name="krona_format" file="mapseq2biom/krona_input.tabular" />
+            <assert_command>
+                <has_text text="-seed '12'" n="0" />
```
Comment (bernt-matthias) [inline tools/mapseq/mapseq.xml:207]:
```suggestion
                <has_text text="-seed" negate="true" />
```

File: tools/mapseq/mapseq.xml [hunk 6]
```diff
@@ -189,13 +203,19 @@
...
+            </assert_command>
         </test>
         <test expect_num_outputs="1">
             <param name="db_source" value="cached" />
             <param name="db_cached" value="test_mapseq_db" />
             <param name="sequences" value="sequences.fasta"/>
             <param name="mapseq2biom" value="no"/>
             <output name="classifications" file="sequences.mapseq" sort="true"/>
+            <assert_command>
+                <has_text text="-seed '12'" n="0" />
```
Comment (bernt-matthias) [inline tools/mapseq/mapseq.xml:217]:
```suggestion
                <has_text text="-seed" negate="true" />
```

File: tools/mapseq/mapseq.xml [hunk 7]
```diff
@@ -206,17 +226,24 @@
             <output name="classifications" file="sequences.mapseq" sort="true"/>
             <output name="otu_tsv" file="mapseq2biom/tab-output_including_taxIDs.tabular" />
             <output name="otu_tsv_notaxid" file="mapseq2biom/tab-output_without_taxIDs.tabular" />
+            <assert_command>
+                <has_text text="-seed '12'" n="0" />
```
Comment (bernt-matthias) [inline tools/mapseq/mapseq.xml:230]:
```suggestion
                <has_text text="-seed" negate="true" />
```

**Review by bernt-matthias — COMMENTED** [review]
Thanks a lot .. just one more minor thing.

**Review by bernt-matthias — APPROVED** [review]

**Pull request merged**